presto-main/src/main/java/com/facebook/presto/sql/planner/NodePartitioningManager.java

```diff
@@ -169,4 +175,17 @@ public NodePartitionMap getNodePartitioningMap(Session session, PartitioningHand
             return bucket;
         });
     }
+
+    private Optional<Map<Integer, Node>> getBucketToNode(Session session, PartitioningHandle partitioningHandle)
+    {
+        checkArgument(!(partitioningHandle.getConnectorHandle() instanceof SystemPartitioningHandle));
+
+        ConnectorNodePartitioningProvider partitioningProvider = partitioningProviders.get(partitioningHandle.getConnectorId().get());
+        checkArgument(partitioningProvider != null, "No partitioning provider for connector %s", partitioningHandle.getConnectorId().get());
+
+        return partitioningProvider.getBucketToNode(
+                partitioningHandle.getTransactionHandle().orElse(null),
+                session.toConnectorSession(),
+                partitioningHandle.getConnectorHandle());
```

Comment:
The 2 lines of assertion should be copied to here instead. Doing so would address dain's comment a few line up as well.

Reply:
@haozhun  Let's talk offline.